.github/workflows/deploy-dev.yml

```diff
@@ -0,0 +1,185 @@
+            ${{ secrets.HOST_DATA }}
+            ${{ secrets.HOST }}
+          private-key: ${{ secrets.SSH_KEY }}
+
+      - name: Copy to Gadi 
+        run: |
+          rsync -e 'ssh -i ${{ steps.ssh.outputs.private-key-path }}' \
+            ${{ env.NAME }}.tar.gz \
+            ${{ secrets.USER }}@${{ secrets.HOST_DATA }}:${{ vars.PRERELEASE_PACK_LOCATION }}
+
+      - name: Deploy to Gadi
+        env:
+          ENVIRONMENT_LOCATION: ${{ vars.PRERELEASE_DEPLOYMENT_LOCATION }}/${{ env.VERSION }}
+          ENVIRONMENT_SYMLINK: ${{ vars.PRERELEASE_DEPLOYMENT_LOCATION }}/dev
+          MODULE_SYMLINK: ${{ vars.PRERELEASE_MODULE_LOCATION }}/dev
+          PACKED_ENVIRONMENT: ${{ vars.PRERELEASE_PACK_LOCATION }}/${{ env.NAME }}.tar.gz
+        run: |
+          ssh ${{ secrets.USER }}@${{ secrets.HOST }} -i ${{ steps.ssh.outputs.private-key-path }} /bin/bash <<'EOT'
+
+          # Create list of previous payu/dev environments to remove later
+          old_versions=$(ls "${{ vars.PRERELEASE_DEPLOYMENT_LOCATION }}" | grep -E '^dev-[0-9]{8}T[0-9]{6}Z-.*')
+        
+          # Unpack conda enviroment
+          mkdir ${{ env.ENVIRONMENT_LOCATION }}
+          if [ $? -ne 0 ]; then
+            exit $?
+          fi
+          tar -xzf ${{ env.PACKED_ENVIRONMENT }} -C ${{ env.ENVIRONMENT_LOCATION }}
+          source ${{ env.ENVIRONMENT_LOCATION }}/bin/activate
+          conda-unpack
+          payu --version
+          source ${{ env.ENVIRONMENT_LOCATION }}/bin/deactivate
+          echo "New payu/dev environment unpacked to ${{ env.ENVIRONMENT_LOCATION }}"
+
+          # Check if payu/dev symlink already exists
+          if [ -L "${{ env.ENVIRONMENT_SYMLINK}}" ]; then
+            # Get previous version  of payu/dev
+            previous_env_path=$(readlink -f "${{ env.ENVIRONMENT_SYMLINK}}")
+            previous_ver=$(basename "$previous_env_path")
+            echo "Previous payu/dev version $previous_ver"
+
+            # Remove previous version from old versions - to ensure active
+            # environments at time of deployment are not deleted
+            old_versions=$(echo "$old_versions" | grep -v "$previous_ver")
+
+            # Unlink the symlink
+            unlink ${{ env.ENVIRONMENT_SYMLINK}}
+          fi
+
+          # Create the payu/dev symlink in pre-release apps
+          ln -s ${{ env.ENVIRONMENT_LOCATION }} ${{ env.ENVIRONMENT_SYMLINK }}
+
+          # Setup modulefile symlink
+          ln -sf ${{ vars.PRERELEASE_MODULE_LOCATION }}/.common ${{ env.MODULE_SYMLINK }}
+
+          # Remove old versions of environments
+          for version in $old_versions; do
+              echo "Removing old payu/dev environment version $version"
+              rm -rf "${{ vars.PRERELEASE_DEPLOYMENT_LOCATION }}/$version"
+          done
+
+          # Remove packed environment file
+          echo "Removing conda-packed environment file ${{ env.PACKED_ENVIRONMENT }}"
+          rm -rf ${{ env.PACKED_ENVIRONMENT }}
```

Comment:
So this wasn't an issue here, but when doing `rm -rf` I've started checking what happens if variables don't evaluate properly (or are null). For example, if the `${{ vars.PRERELEASE_PACKED_LOCATION }}` wasn't in the `Gadi` environment, it would evaluate to `rm -rf /payu-1.1.tar.gz` which isn't disastrous. 

But if `${{ env.PACKED_ENVIRONMENT }}` was something like `PACKED_ENVIRONMENT: ${{ env.PRERELEASE_PACKED_LOCATION }}/${{ env.NAME }}` (where you include the `.tar.gz` in the name) and both are evaluatable to null, one could have `rm -rf /` on Gadi, which is a scary thought...

Reply:
Yeah that it is a scary thought.. Ok so good in that there's a `.tar.gz` at the moment..